FLockDataset/utils/chain.py

```
@@ -18,4 +21,101 @@ def assert_registered(wallet: bt.wallet, metagraph: bt.metagraph) -> int:
+            data = json.loads(json_str)
+        except json.JSONDecodeError as e:
+            print("Failed to parse JSON:", e)
+            return None
+
+        comp = Competition.from_dict(data)
+        print("🔍 Competition.from_dict →", comp)
+        return comp
 
+    except Exception as e:
+        print("Unhandled exception in read_chain_commitment:", e)
+        import traceback; traceback.print_exc()
```

Ruff (0.8.2)

Multiple statements on one line (semicolon)

(E702)